chucks-pick3-web/angular-app/src/app/components/demo.component/demo.component.ts

```diff
@@ -7,10 +10,21 @@ import {Component, OnInit } from '@angular/core';
 })
 
 export class DemoComponent implements OnInit {
-  constructor() {
+  constructor(private subNavBarService: SubNavBarService,
+              private route: ActivatedRoute) {
   }
 
   ngOnInit() {
 
-  }
-}
+    if (this.route.snapshot.queryParamMap.get('showSubNavBar') === null) {
+      stop();
+
+
+    } else {
+      this.route.queryParamMap.pipe(map(params => {
+        return params.get('showSubNavBar').toLowerCase() === 'true' ? true : false;
```

Comment:
Need to check if params is null or undefined first.  Only then can you call the `params.get('showSubNavBar')` method.  At which point this method call could return null or undefined.  You will need to do another check for that.  Only then can you call `.toLowerCase()` method.

Update this code and all other code in appropriate components.

Reply:
I've updated the code using 
 if (this.route.snapshot.queryParamMap.get('showSubNavBar') === null || undefined) {

I debugged it and did not run into issues.

Reply:
I created validation checks for both the params and get(showSubNavBar).